class.jetpack-sync.php

```diff
@@ -1,1104 +0,0 @@
-	 * Cannot be called statically
-	 *
-	 * @param int $id Post ID
-	 * @return Array containing full post details
-	 */
-	function get_post( $id ) {
-		$post_obj = get_post( $id );
-		if ( !$post_obj )
-			return false;
-
-		if ( is_callable( $post_obj, 'to_array' ) ) {
-			// WP >= 3.5
-			$post = $post_obj->to_array();
-		} else {
-			// WP < 3.5
-			$post = get_object_vars( $post_obj );
-		}
-
-		if ( 0 < strlen( $post['post_password'] ) ) {
-			$post['post_password'] = 'auto-' . wp_generate_password( 10, false ); // We don't want the real password.  Just pass something random.
-		}
-
-		// local optimizations
-		unset(
-			$post['filter'],
-			$post['ancestors'],
-			$post['post_content_filtered'],
-			$post['to_ping'],
-			$post['pinged']
-		);
-
-		if ( $this->is_post_public( $post ) ) {
-			$post['post_is_public'] = Jetpack_Options::get_option( 'public' );
-		} else {
-			//obscure content
-			$post['post_content'] = '';
-			$post['post_excerpt'] = '';
-			$post['post_is_public'] = false;
-		}
-		$post_type_obj = get_post_type_object( $post['post_type'] );
-		$post['post_is_excluded_from_search'] = $post_type_obj->exclude_from_search;
-
-		$post['tax'] = array();
-		$taxonomies = get_object_taxonomies( $post_obj );
-		foreach ( $taxonomies as $taxonomy ) {
-			$terms = get_object_term_cache( $post_obj->ID, $taxonomy );
-			if ( empty( $terms ) )
-				$terms = wp_get_object_terms( $post_obj->ID, $taxonomy );
-			$term_names = array();
-			foreach ( $terms as $term ) {
-				$term_names[] = $term->name;
-			}
-			$post['tax'][$taxonomy] = $term_names;
-		}
-
-		$meta = get_post_meta( $post_obj->ID, false );
-		$post['meta'] = array();
-		foreach ( $meta as $key => $value ) {
-			$post['meta'][$key] = array_map( 'maybe_unserialize', $value );
-		}
-
-		$post['extra'] = array(
-			'author' => get_the_author_meta( 'display_name', $post_obj->post_author ),
-			'author_email' => get_the_author_meta( 'email', $post_obj->post_author ),
-			'dont_email_post_to_subs' => get_post_meta( $post_obj->ID, '_jetpack_dont_email_post_to_subs', true ),
-		);
-
-		if ( $fid = get_post_thumbnail_id( $id ) ) {
-			$feature = wp_get_attachment_image_src( $fid, 'large' );
-			if ( ! empty( $feature[0] ) ) {
-				$post['extra']['featured_image'] = $feature[0];
-			}
-
-			$attachment = get_post( $fid );
-			if ( ! empty( $attachment ) ) {
-				$metadata = wp_get_attachment_metadata( $fid );
-
-				$post['extra']['post_thumbnail'] = array(
-					'ID'        => (int) $fid,
-					'URL'       => (string) wp_get_attachment_url( $fid ),
-					'guid'      => (string) $attachment->guid,
-					'mime_type' => (string) $attachment->post_mime_type,
-					'width'     => (int) isset( $metadata['width'] ) ? $metadata['width'] : 0,
-					'height'    => (int) isset( $metadata['height'] ) ? $metadata['height'] : 0,
-				);
-
-				if ( isset( $metadata['duration'] ) ) {
-					$post['extra']['post_thumbnail'] = (int) $metadata['duration'];
-				}
-
-				/**
```

Comment:
You'll need to move the documentation to this filter to avoid losing documentation in our codex:
https://github.com/Automattic/jetpack/blob/master/json-endpoints/class.wpcom-json-api-post-endpoint.php#L633